sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-calculation/e-cert-calculation-process.ts

```diff
@@ -136,4 +148,39 @@ export abstract class ECertCalculationProcess {
       }
     }
   }
+
+  /**
+   * Creates e-cert blocked notification.
+   * @param studentId student id associated with the blocked disbursement.
+   * @param applicationNumber application number associated with the blocked disbursement.
+   * @param entityManager entity manager to execute in transaction.
+   */
+  private async createEcertBlockedNotification(
+    studentId: number,
+    applicationNumber: string,
+    entityManager: EntityManager,
+  ) {
+    const auditUser = this.systemUsersService.systemUser;
+    const student = await entityManager.getRepository(Student).findOne({
+      select: {
+        id: true,
+        birthDate: true,
+        user: { firstName: true, lastName: true, email: true },
+      },
+      relations: { user: true },
+      where: { id: studentId },
+    });
+    const notification = {
+      givenNames: student.user.firstName,
+      lastName: student.user.lastName,
+      email: student.user.email,
+      dob: new Date(student.birthDate),
+      applicationNumber: applicationNumber,
+    };
+    await this.notificationActionsService.saveEcertBlockedNotification(
```

Comment:
I would not recommend to use one notification method to save student and ministry notification both.

I would advice to create separate methods. `saveBlockedStudentNotification` and `saveBlockedMinistryNotification`